driver/compplugin.ml

```diff
@@ -0,0 +1,51 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*         Fabrice Le Fessant, projet Gallium, INRIA Rocquencourt         *)
+(*                                                                        *)
+(*   Copyright 1996 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(* A table to avoid double linking of plugins, especially with OCAMLPARAM *)
+let plugins = Hashtbl.create 13
+
+let load plugin_name =
+
+  try
+    let plugin_name = Compdynlink.adapt_filename plugin_name in
+
+    let plugin_file =
+      if Filename.is_implicit plugin_name then
+        try
+          Compmisc.init_path !Clflags.native_code;
+          Misc.find_in_path !Config.load_path plugin_name
+        with Not_found ->
+          raise (Compdynlink.Error (Compdynlink.File_not_found plugin_name))
+      else plugin_name
+    in
+
+    try
+
+      if not (Hashtbl.mem plugins plugin_file) then begin
+        Compdynlink.loadfile plugin_file;
+        Hashtbl.add plugins plugin_file (); (* plugin loaded *)
+      end
+    with Compdynlink.Error error ->
+      Printf.eprintf "Error while loading plugin file %S:\n  %s\n%!"
+        plugin_file (Compdynlink.error_message error);
+      exit 2
+
+  with Compdynlink.Error error ->
```

Comment:
Can this second handler catch anything else than the explicit `File_not_found` raise above?

In any case, the two handlers could be merged, no?

And I'd prefer to raise a different error (with a properly registered printer) rather than stop the process.